crypto/nrf_cc310_platform/src/nrf_cc310_platform_mutex_zephyr.c

```diff
@@ -0,0 +1,242 @@
+
+/** @brief External reference to the platforms abort APIs
+ *  	   This is used in case the mutex functions don't
+ * 		   provide return values in their APIs.
+ */
+extern nrf_cc310_platform_abort_apis_t platform_abort_apis;
+
+/** @brief Definition of mutex for symmetric cryptography
+ */
+K_MUTEX_DEFINE(sym_mutex_int);
+
+/** @brief Definition of mutex for asymmetric cryptography
+ */
+K_MUTEX_DEFINE(asym_mutex_int);
+
+/** @brief Definition of mutex for random number generation
+*/
+K_MUTEX_DEFINE(rng_mutex_int);
+
+/** @brief Definition of mutex for threading operations
+*/
+K_MUTEX_DEFINE(thread_mutex_int);
+
+/** @brief Definition of mutex for power mode changes
+*/
+K_MUTEX_DEFINE(power_mutex_int);
+
+#define NUM_MUTEXES 64
+
+struct k_mem_slab mutex_slab;
+char __aligned(4) mutex_slab_buffer[NUM_MUTEXES * sizeof(struct k_mutex)];
+
+
+/**@brief Definition of RTOS-independent symmetric cryptography mutex
+ * with NRF_CC310_PLATFORM_MUTEX_MASK_IS_VALID set to indicate that
+ * allocation is unneccesary
+*/
+nrf_cc310_platform_mutex_t sym_mutex =
+{
+	.mutex = &sym_mutex_int,
+	.flags = NRF_CC310_PLATFORM_MUTEX_MASK_IS_VALID
+};
+
+/**@brief Definition of RTOS-independent asymmetric cryptography mutex
+ * with NRF_CC310_PLATFORM_MUTEX_MASK_IS_VALID set to indicate that
+ * allocation is unneccesary
+*/
+nrf_cc310_platform_mutex_t asym_mutex =
+{
+	.mutex = &asym_mutex_int,
+	.flags = NRF_CC310_PLATFORM_MUTEX_MASK_IS_VALID
+};
+
+
+/**@brief Definition of RTOS-independent random number generation mutex
+ * with NRF_CC310_PLATFORM_MUTEX_MASK_IS_VALID set to indicate that
+ * allocation is unneccesary
+*/
+nrf_cc310_platform_mutex_t rng_mutex =
+{
+	.mutex = &rng_mutex_int,
+	.flags = NRF_CC310_PLATFORM_MUTEX_MASK_IS_VALID
+};
+
+
+/**@brief Definition of RTOS-independent threading mutex
+ * with NRF_CC310_PLATFORM_MUTEX_MASK_IS_VALID set to indicate that
+ * allocation is unneccesary
+*/
+nrf_cc310_platform_mutex_t thread_mutex =
+{
+	.mutex = &thread_mutex_int,
+	.flags = NRF_CC310_PLATFORM_MUTEX_MASK_IS_VALID
+};
+
+
+/**@brief Definition of RTOS-independent power management mutex
+ * with NRF_CC310_PLATFORM_MUTEX_MASK_IS_VALID set to indicate that
+ * allocation is unneccesary
+*/
+nrf_cc310_platform_mutex_t power_mutex =
+{
+	.mutex = &power_mutex_int,
+	.flags = NRF_CC310_PLATFORM_MUTEX_MASK_IS_VALID
+};
+
+/**@brief static function to initialize a mutex
+ */
+static void mutex_init(nrf_cc310_platform_mutex_t *mutex) {
+
+	/* Ensure that the mutex is valid (not NULL) */
+	if (mutex == NULL) {
+		platform_abort_apis.abort_fn("mutex_init called with NULL parameter");
+	}
+
+	/* Allocate if this has not been initialized statically */
+	if (mutex->flags == NRF_CC310_PLATFORM_MUTEX_MASK_INVALID &&
+	    mutex->mutex == NULL) {
+		/* allocate some memory for the mute*/
+		k_mem_slab_alloc(&mutex_slab, &mutex->mutex, K_FOREVER);
+		memset(mutex->mutex, 0, sizeof(struct k_mutex));
+		if (mutex->mutex == NULL)
```

Comment:
Wouldn't you want to check this before you memset it?

Reply:
Will fix